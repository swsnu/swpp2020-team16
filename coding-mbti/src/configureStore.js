import {
  persistStore,
  persistReducer,
  FLUSH,
  REHYDRATE,
  REGISTER,
  PAUSE,
  PERSIST,
  PURGE,
} from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import reducer from './feature';

const persistConfig = {
  key: 'root',
  storage,
};

const persistedReducer = persistReducer(persistConfig, reducer);

export default () => {
  const store = configureStore({
    reducer: persistedReducer,
    middleware: getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
      },
    }),
  });
  const persistor = persistStore(store);
  return { store, persistor };
};
