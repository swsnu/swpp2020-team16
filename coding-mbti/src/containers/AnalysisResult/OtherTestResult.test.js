import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';

import { ThemeProvider, createMuiTheme } from '@material-ui/core';
import { transitions, positions, Provider as AlertProvider } from 'react-alert';
import AlertTemplate from 'react-alert-template-basic';
import configureStore from '../../configureStore';
import OtherTestResult from './OtherTestResult';

const { store } = configureStore();
const theme = createMuiTheme();

describe('<OtherTestResult/>', () => {
    let otherSolutions;

    beforeEach(() => {
        const options = {
            // you can also just use 'bottom center'
            position: positions.BOTTOM_CENTER,
            timeout: 5000,
            offset: '30px',
            // you can also just use 'scale'
            transition: transitions.SCALE,
        };

        otherSolutions = (
            <Provider store={store}>
                <ThemeProvider theme={theme}>
                    <AlertProvider template={AlertTemplate} {...options}>
                        <OtherTestResult />
                    </AlertProvider>
                </ThemeProvider>
            </Provider>
        );
    });

    it('should render withour any error', () => {
        const component = mount(otherSolutions);
        expect(component).toMatchSnapshot();
    });
});
