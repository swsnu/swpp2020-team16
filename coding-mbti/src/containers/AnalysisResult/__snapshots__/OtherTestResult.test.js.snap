// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OtherTestResult/> should render withour any error 1`] = `ReactWrapper {}`;
